### PR TITLE
Fixed #1527 to #1529

### LINKED ISSUE #1527: Water Tank gives StackOverflowError after right clicked on.
Hi,

I tried to build a Water Tank but at the first try nothing happened by right clicking it.
Then I broke some blocks of the structure, I placed them again and I right clicked and I got a crash that occurs every time I try to play on the world.

I'm using Railcraft on development environment (GitHub, mc-1.12.2).

Here's the crash report: [crash-2018-11-03_11.12.34-server.txt](https://github.com/Railcraft/Railcraft/files/2544943/crash-2018-11-03_11.12.34-server.txt)

Thank you.
Axelandre

### PR DESCRIPTION
**Pull request by liach**
#1527 wrong reference to standalone inventory
#1528 edge cases for isMaster not updated when master block is set to null
#1529 null check for tagged cart on coupler

### CODE REVIEW

File: src/main/java/mods/railcraft/common/blocks/multi/TileMultiBlock.java [hunk 1]
```diff
@@ -454,6 +454,7 @@ public final boolean isStructureValid() {
     @Override
     public final @Nullable TileMultiBlock getMasterBlock() {
         if (masterBlock != null && !isStructureValid()) {
+            masterBlock.isMaster = false;
```
Comment (CovertJaguar) [inline src/main/java/mods/railcraft/common/blocks/multi/TileMultiBlock.java]: Don't like this. Its not the slaves job to determine whether a master is a master.

**Review by CovertJaguar — CHANGES_REQUESTED** [review]

File: src/main/java/mods/railcraft/common/blocks/multi/TileMultiBlock.java (hunk 1 above)
Comment (liach) [reply src/main/java/mods/railcraft/common/blocks/multi/TileMultiBlock.java]: So should I call `isStructureValid()` in `isMaster()` check?

Comment (liach) [reply src/main/java/mods/railcraft/common/blocks/multi/TileMultiBlock.java]: Updated. Added a contract to call both methods if work should be done.

**Review by CovertJaguar — APPROVED** [review]

**Comment by CovertJaguar** [comment]
This seems safer, but I'm still thinking I need to completely redesign the logic for multiblocks so this stuff needs less boilerplate.

**Comment by liach** [comment]
thanks!

**Pull request merged**